lighthouse-core/report/html/renderer/report-renderer.js

```diff
@@ -193,7 +193,14 @@ class ReportRenderer {
     Util.i18n = i18n;
     Util.reportJson = report;
 
-    const detailsRenderer = new DetailsRenderer(this._dom);
+    const fullPageScreenshot = /** @type {LH.Artifacts.FullPageScreenshot | undefined} */ (
```

Comment:
same thing with the audit details type and possibly avoiding the cast

Reply:
how to avoid the cast?

Reply:
> how to avoid the cast?

caveating that I could be wrong because I haven't tried it :) I assume `details` will already either be `LH.Audit.Details` or `undefined`, so testing that it exists and has `type` `full-page-screenshot` should be enough to convince tsc.

Reply:
alright how's that

Reply:
for `report-ui-features.js` could also do something like
```js
const fullPageScreenshot = report.audits['full-page-screenshot'] && report.audits['full-page-screenshot'].details;
if (fullPageScreenshot || fullPageScreenshot.type !== 'full-page-screenshot') return;
```
or some kind of code golfing, but it's also good enough if you're happy :)

Reply:
cool, just clocks in at 100 characters 
![image](https://user-images.githubusercontent.com/4071474/87835749-9bcd8680-c853-11ea-8341-ba7d170aa048.png)